### DISCUSSION TITLE
Train Faster-RCNN on Cityscapes

### DISCUSSION THREAD

**Discussion by pinxau1000** [facebookresearch/detectron2#3990] [Q&A]
Hello all,

I have run [Mask R-CNN](https://github.com/facebookresearch/detectron2/blob/main/configs/Cityscapes/mask_rcnn_R_50_FPN.yaml) on [cityscapes](https://www.cityscapes-dataset.com/) dataset without any issue, however now I want to run [Faster R-CNN](https://github.com/facebookresearch/detectron2/blob/main/configs/COCO-Detection/faster_rcnn_R_50_FPN_3x.yaml) on the same dataset.

I'm using as reference the paper `On Intra Video Coding And In-Loop Filtering For Neural Object Detection Networks`  where the authors state:
> The same (Mask R-CNN) backbone is taken for Faster R-CNN, but since there is no pre-trained Cityscapes model available in Detectron2, the already existing model trained on the COCO dataset is taken from Detectron2 as initial weights. Subsequently, this model is further trained on the Cityscapes classes and training images, where the GT pixel masks have been converted to bounding boxes before, for 42000 iterations (...)

**I'm asking for ideas for implementing this approach, i.e. re-train Faster R-CNN on Cityscapes dataset.**

###### Note: I tried to use [train_net.py](https://github.com/facebookresearch/detectron2/blob/main/tools/train_net.py) with Faster R-CNN model on Cityscapes dataset and the error below was been thrown. I also tried to implement a [custom dataloader](https://detectron2.readthedocs.io/en/latest/tutorials/datasets.html#use-custom-datasets) and a [custom evaluator](https://detectron2.readthedocs.io/en/latest/tutorials/evaluation.html#evaluation) based on [plain_train_net.py](https://github.com/facebookresearch/detectron2/blob/main/tools/plain_train_net.py) which produced the same exception.
``` python
File ".venv/lib/python3.8/site-packages/detectron2/modeling/proposal_generator/rpn.py", line 470, in forward
    assert gt_instances is not None, "RPN requires gt_instances in training!"
AssertionError: RPN requires gt_instances in training!
```

Best regards
pinxau1000

**Comment by pinxau1000** [comment]
@ppwwyyxx Can you help me out?

**Reply by Chris1nexus** [reply]
Have you set  the default MODEL.MASK_ON = False 
to MODEL.MASK_ON = True   ?
I had the same issue some time ago due to this default setting on the detectron2 cfg file
